examples/documentation_examples/examples.go

```diff
@@ -2097,3 +2097,645 @@ func ChangeStreamExamples(t *testing.T, db *mongo.Database) {
+					{"_id", 0},
+					{"name", 1},
+					{"airlines", bson.D{
+						{"$filter", bson.D{
+							{"input", "$airlines"},
+							{"as", "airline"},
+							{"cond", bson.D{
+								{"$eq", bson.A{"$$airline.country", "Canada"}},
+							}},
+						}},
+					}},
+				}},
+			},
+		}
+
+		cursor, err := airAlliancesColl.Aggregate(ctx, pipeline)
+
+		// End Aggregation Example 7
+
+		require.NoError(t, err)
+		defer cursor.Close(ctx)
+
+		ok := cursor.Next(ctx)
+
+		require.True(t, ok)
+	}
+}
+
+// RunCommandExamples contains examples of RunCommand operations.
+func RunCommandExamples(t *testing.T, db *mongo.Database) {
+	ctx := context.Background()
+
+	coll := db.Collection("restaurants")
+
+	{
+		// Start RunCommand Example 1
+		docs := []interface{}{
+			bson.D{
+				{"name", "Chez Panisse"},
+				{"city", "Oakland"},
+				{"state", "California"},
+				{"country", "United States"},
+				{"rating", 4.4},
+			},
+			bson.D{
+				{"name", "Central"},
+				{"city", "Lima"},
+				{"country", "Peru"},
+				{"rating", 4.8},
+			},
+			bson.D{
+				{"name", "Eleven Madison Park"},
+				{"city", "New York City"},
+				{"state", "New York"},
+				{"country", "United States"},
+				{"rating", 4.6},
+			},
+			bson.D{
+				{"name", "Gaggan"},
+				{"city", "Bangkok"},
+				{"country", "Thailand"},
+				{"rating", 4.3},
+			},
+			bson.D{
+				{"name", "Dad's Grill"},
+				{"city", "Oklahoma City"},
+				{"state", "Oklahoma"},
+				{"country", "United States"},
+				{"rating", 2.1},
+			},
+		}
+
+		result, err := coll.InsertMany(ctx, docs)
+
+		// End RunCommand Example 1
+
+		require.NoError(t, err)
+		require.Len(t, result.InsertedIDs, 5)
+	}
+	{
+		// Start RunCommand Example 2
+		res := db.RunCommand(ctx, bson.D{{"buildInfo", 1}})
+
+		// End RunCommand Example 2
+
+		err := res.Err()
+		require.NoError(t, err)
+	}
+	{
+		// Start RunCommand Example 3
+		res := db.RunCommand(ctx, bson.D{{"collStats", "restaurants"}})
+
+		// End RunCommand Example 3
+
+		err := res.Err()
+		require.NoError(t, err)
+	}
+}
+
+// IndexExamples contains examples of Index operations.
+func IndexExamples(t *testing.T, db *mongo.Database) {
+	ctx := context.Background()
+
+	recordsColl := db.Collection("records")
+	restaurantsColl := db.Collection("restaurants")
+	{
+		// Start Index Example 1
+		docs := []interface{}{
+			bson.D{
+				{"student", "Marty McFly"},
+				{"classYear", 1986},
+				{"school", "Hill Valley High"},
+				{"GPA", 2.1},
+			},
+			bson.D{
+				{"student", "Ferris F. Bueller"},
+				{"classYear", 1987},
+				{"school", "Glenbrook North High"},
+				{"status", "Suspended"},
+				{"GPA", 3.4},
+			},
+			bson.D{
+				{"student", "Reynard Muldoon"},
+				{"classYear", 1993},
+				{"school", "Stonetown Middle"},
+				{"GPA", 4.0},
+			},
+		}
+
+		result, err := recordsColl.InsertMany(ctx, docs)
+
+		// End Index Example 1
+
+		require.NoError(t, err)
+		require.Len(t, result.InsertedIDs, 3)
+	}
+	{
+		// Start Index Example 2
+		docs := []interface{}{
```

Comment:
Can we move this `InsertMany` call to example 1 so all of the setup is isolated?

Reply:
Yep! See above comment